maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -1152,8 +1173,11 @@ private SurefireProperties setupProperties() {
             }
         }
 
-        SurefireProperties result = SurefireProperties.calculateEffectiveProperties(
-                getSystemProperties(), getSystemPropertyVariables(), getUserProperties(), sysProps);
+        SurefireProperties result = calculateEffectiveProperties(
+                getSystemProperties(),
+                getSystemPropertyVariables(),
+                useUserPropertiesAsSystemProperty ? getUserProperties() : null,
```

Comment:
Wouldn't it be better, for consistency reasons, to pass empty properties here instead of `null`. That makes the code more consistent with the other sources.

Reply:
Done in https://github.com/apache/maven-surefire/pull/762/commits/f21ac963663d27d8b316d789648cb6da44088e4d.